lib/Cache.js

```diff
@@ -125,25 +128,27 @@ class Cache {
 				if (typeof value === 'undefined') { continue; }
 
 				for (const prop in slice.filter) {
+
 					if (
 						!this._checkFilter(entity, prop, slice.filter[prop])
 					) {
 						continue keyloop; //eslint-disable-line
 					}
 				}
-				if (this.knowledge[key].contains(value)) {
-					keyValues[key] = { value, item: entity };
-				}
 
-				if (slice.link) { for (const linkType in slice.link) {
-					const dKey = this.getDualKey(slice, linkType);
+				keyValues[key] = { value, item: entity };
```

Comment:
If the current knowledge is say [ [1, 5] ] and we add an item with order value 7?
or if we update the order property of  an item from 3 to 6.

The value won't be in the range. So we would want to add a new range and change it to something like [ [1, 5], [6] ].